src/utils/hooks/CartContext.js

```diff
@@ -0,0 +1,62 @@
+import React, { createContext, useContext, useMemo, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import {
+  ADD_ITEM,
+  cartInitialState,
+  cartReducer,
+  EMPTY_CART,
+  REMOVE_ITEM,
+  SET_ITEM_QUANTITY,
+} from '../reducers/cartReducer';
+
+const CartContext = createContext({
+  cart: {},
+  addItemToCart: () => {},
+  removeItemFromCart: () => {},
+});
+
+export function CartContextProvider({ children }) {
+  const [cartState, dispatch] = useReducer(cartReducer, cartInitialState);
+
+  const addProductToCart = (product, quantity) => {
+    dispatch({ type: ADD_ITEM, product, quantity });
+  };
+
+  const setProductCountToCart = (product, quantity) => {
```

Comment:
Please rename it to `update...` the set word is not the appropriate one for this scenario.